### PR TITLE
sql: add fast mode

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Add fast mode

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/10560
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-read-in-fast-mode.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+summary: Introduces a way to speed up the querying of AP scenarios by using Fast Mode.
+---
+
+# Fast Mode
+
+This document describes how to use Fast Mode to speed up queries in Analytical Processing (AP) scenarios.
+
+TiFlash supports the following modes:
+
+- `Normal Mode` mode. The default mode. This mode guarantees the accuracy of query results and data consistency.
```
Comment (hongyunyan) [inline develop/dev-guide-read-in-fast-mode.md:12]:
```suggestion
- `Normal Mode`. The default mode. This mode guarantees the accuracy of query results and data consistency.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+- `Fast Mode` mode. This mode does not guarantee the accuracy of query results and data consistency, but provides more efficient query performance.
```
Comment (hongyunyan) [inline develop/dev-guide-read-in-fast-mode.md:13]:
```suggestion
- `Fast Mode`. This mode does not guarantee the accuracy of query results and data consistency, but provides more efficient query performance.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+AP allows for some tolerance to the accuracy of the query results. If you need higher query performance, you can switch the corresponding table to TiFlash's Fast Mode for querying.
```
Comment (hongyunyan) [inline develop/dev-guide-read-in-fast-mode.md:15]:
```suggestion
Some AP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can switch the corresponding table to TiFlash's Fast Mode for querying.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+You can switch back to Normal Mode using the following statement.
+
+```sql
+ALTER TABLE table_name SET TIFALSH MODE NORMAL
+```
+
+## Mechanism of Fast Mode
+
+The data in the stroage layer of TiFlash is stored in two layers: Delta layer and Stable layer.
```
Comment (hongyunyan) [inline develop/dev-guide-read-in-fast-mode.md:47]: stroage --> storage
I think we may can change the expression of 'storage layer'. Too many 'layers' may make a little weird.

File: develop/dev-guide-read-in-fast-mode.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+The overall TableScan algorithm process in Normal Mode consists of the following steps:
+
+1. Read data: separate data streams are created in the Delta layer and Stable layer to read the respective data.
+2. Sort Merge: the data streams created in step 1 are merged and the data is returned in (handle, version) order.
```
Comment (hongyunyan) [inline develop/dev-guide-read-in-fast-mode.md:52]:
```suggestion
2. Sort Merge: the data streams created in step 1 are sorted merged and the data is returned in (handle, version) order.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+3. Range Filter: the data from step 2 is filtered and returned.
```
Comment (hongyunyan) [inline develop/dev-guide-read-in-fast-mode.md:53]: I think the descriptions of these two steps(3,4) are not match the description in chinese articles, maybe need a change.

File: develop/dev-guide-read-in-fast-mode.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+4. MVCC + Column Filter: the data from step 3 is filtered by MVCC, and the unneeded columns are filtered out and the the data is returned.
+
+Fast Mode mode gains faster query performance by sacrificing some data consistency. The TableScan process in Fast Mode omits Step 2 and the MVCC part in Step 4 in the Normal Mode process, thus improving query performance.
```
Comment (hongyunyan) [inline develop/dev-guide-read-in-fast-mode.md:56]:
```suggestion
Fast Mode gains faster query performance by sacrificing some data consistency. The TableScan process in Fast Mode omits Step 2 and the MVCC part in Step 4 in the Normal Mode process, thus improving query performance.
```

File: sql-statements/sql-statement-set-tiflash-mode.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+# `ALTER TABLE ... SET TIFLASH MODE ...`
+
+> **Warning:**
+>
+> This statement is still an experimental feature. It is NOT recommended that you use it in the production environment.
+
+You can use the `ALTER TABLE ... SET TIFLASH MODE ... ` statement to switch the mode of the corresponding table in TiFlash. The following modes are currently supported.
+
+- `Normal Mode` mode. The default mode. This mode guarantees the accuracy of query results and data consistency.
```
Comment (hongyunyan) [inline sql-statements/sql-statement-set-tiflash-mode.md:14]:
```suggestion
- `Normal Mode`. The default mode. This mode guarantees the accuracy of query results and data consistency.
```

File: sql-statements/sql-statement-set-tiflash-mode.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+- `Fast Mode` mode. This mode does not guarantee the accuracy of query results and data consistency, but provides more efficient query performance.
```
Comment (hongyunyan) [inline sql-statements/sql-statement-set-tiflash-mode.md:15]:
```suggestion
- `Fast Mode`. This mode does not guarantee the accuracy of query results and data consistency, but provides more efficient query performance.
```

File: sql-statements/sql-statement-set-tiflash-mode.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+This statement executes without blocking the execution of existing SQL statements or the running of TiDB features, such as transactions, DDL, and GC, and without changing the data content accessed through the SQL statement. The statement will end normally when the schema switch is completed.
```
Comment (hongyunyan) [inline sql-statements/sql-statement-set-tiflash-mode.md:17]:
```suggestion
This statement executes without blocking the execution of existing SQL statements or the running of TiDB features, such as transactions, DDL, and GC, and without changing the data content accessed through the SQL statement. The statement will end normally when the mode switch is completed.
```

File: develop/dev-guide-read-in-fast-mode.md (hunk 4 above)
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:47]:
```suggestion
The data in the storage layer of TiFlash is stored in two layers: Delta layer and Stable layer.
```

Comment (hfxsd) [reply develop/dev-guide-read-in-fast-mode.md:47]: fixed

File: develop/dev-guide-read-in-fast-mode.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+## Mechanism of Fast Mode
+
+The data in the storage layer of TiFlash is stored in two layers: Delta layer and Stable layer.
+
+The overall TableScan algorithm process in Normal Mode consists of the following steps:
+
+1. Read data: separate data streams are created in the Delta layer and Stable layer to read the respective data.
+2. Sort Merge: the data streams created in step 1 are sorted and merged, and the data is returned in (handle, version) order.
+3. Range Filter: the data from step 2 is filtered and returned.
```
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:53]:
```suggestion
3. Range Filter: filters the data generated in step 2 and returns it.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+4. MVCC + Column Filter: the data from step 3 is filtered by MVCC, and the unneeded columns are filtered out and the the data is returned.
```
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:54]:
```suggestion
4. MVCC + Column Filter: filter the data generated in step 3 through MVCC. Meanwhile, filter out the unneeded columns and the the data is returned.
```

**Review by hongyunyan — APPROVED** [review]
LGTM

File: develop/dev-guide-read-in-fast-mode.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+This document describes how to use Fast Mode to speed up queries in Analytical Processing (AP) scenarios.
+
+TiFlash supports the following modes:
+
+- `Normal Mode`. The default mode. This mode guarantees the accuracy of query results and data consistency.
+- `Fast Mode`. This mode does not guarantee the accuracy of query results and data consistency, but provides more efficient query performance.
+
+Some AP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can switch the corresponding table to TiFlash's Fast Mode for querying.
+
+The mode switch takes effect globally. TiFlash mode switching is only available for tables with TiFlash Replica. TiFlash-related operations are not supported for temporary tables, in-memory tables, system tables, and tables with non-utf-8 characters in the column names, and therefore changing TiFlash table mode is not supported for them.
```
Comment (joey-yez) [inline develop/dev-guide-read-in-fast-mode.md:17]: The mode switch  takes effect globally only for the table whose mode are set to fast by the sql ’alter table set tiflash mode‘, not all the tiflash tables. 
Make sure users will not be confused.

File: develop/dev-guide-read-in-fast-mode.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:21]:
```suggestion
The mode switch takes effect globally only for those tables whose mode is set to Fast Mode by [ALTER TABLE SET TIFLASH MODE](/sql-statements/sql-statement-set-tiflash-mode.md). TiFlash-related operations are not supported for temporary tables, in-memory tables, system tables, and tables with non-utf-8 characters in the column names, and therefore changing TiFlash table mode is not supported for them.
```

File: develop/dev-guide-read-in-fast-mode.md (hunk 13 above)
Comment (hfxsd) [reply develop/dev-guide-read-in-fast-mode.md:21]: OK. Refined as follows. PTAL.

File: sql-statements/sql-statement-set-tiflash-mode.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+{{< copyable "sql" >}}
+
+```sql
+USE TEST;
+CREATE TABLE test (a INT NOT NULL, b INT);
+ALTER TABLE test SET TIFLASH REPLICA 1;
+```
+
+The default mode of the `test` table is Normal Mode. You can query it with the following statement.
```
Comment (ran-huang) [inline sql-statements/sql-statement-set-tiflash-mode.md:44]:
```suggestion
The default mode of the `test` table is Normal Mode. You can query the table mode with the following statement.
```

File: sql-statements/sql-statement-set-tiflash-mode.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+ALTER TABLE test SET tiflash mode NORMAL
+```
+
+## MySQL compatibility
+
+`ALTER TABLE ... SET TiFLASH MODE ...` is an extension to the standard SQL syntax introduced by TiDB. Although there is no equivalent MySQL syntax, you can still execute this statement from a MySQL client, or from database drivers that follow the MySQL protocol.
+
+## TiDB Binlog and TiCDC compatibility
+
+When the downstream is also TiDB, `ALTER TABLE ... SET TiFLASH MODE ...` will be synchronized downstream by TiDB Binlog. In other scenarios, neither TiDB Binlog nor TiCDC will synchronize this statement.
```
Comment (ran-huang) [inline sql-statements/sql-statement-set-tiflash-mode.md:85]:
```suggestion
When the downstream is also TiDB, `ALTER TABLE ... SET TiFLASH MODE ...` will be synchronized downstream by TiDB Binlog. In other scenarios, TiDB Binlog does not synchronize this statement.

Fast mode does not support TiCDC.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+```sql
+SELECT table_mode FROM information_schema.tiflash_replica WHERE table_name = 'table_name' AND table_schema = 'database_name'
+```
+
+Use the following statement to switch the corresponding table to Fast Mode.
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE table_name SET TIFALSH MODE FAST
```
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:38]:
```suggestion
ALTER TABLE table_name SET TIFLASH MODE FAST
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+
+> **Warning:**
+>
+> This feature is still an experimental feature. It is NOT recommended that you use it in the production environment.
+
+This document describes how to use Fast Mode to speed up queries in Analytical Processing (AP) scenarios.
+
+TiFlash supports the following modes:
+
+- `Normal Mode`. The default mode. This mode guarantees the accuracy of query results and data consistency.
```
[the remark this answers is not in the document]
Comment (hfxsd) [reply develop/dev-guide-read-in-fast-mode.md:16]: ok

File: develop/dev-guide-read-in-fast-mode.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+summary: Introduces a way to speed up the querying of AP scenarios by using Fast Mode.
+---
+
+# Fast Mode
+
+> **Warning:**
+>
+> This feature is still an experimental feature. It is NOT recommended that you use it in the production environment.
+
+This document describes how to use Fast Mode to speed up queries in Analytical Processing (AP) scenarios.
```
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:12]:
```suggestion
This document describes how to use Fast Mode to speed up queries in Online Analytical Processing (OLAP) scenarios.
```

File: develop/dev-guide-read-in-fast-mode.md (hunk 18 above)
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:16]:
```suggestion
- `Normal Mode`. The default mode. This mode guarantees the precision of query results and data consistency.
```

**Review by hongyunyan — APPROVED** [review]

File: develop/dev-guide-read-in-fast-mode.md [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+## Mechanism of Fast Mode
+
+The data in the storage layer of TiFlash is stored in two layers: Delta layer and Stable layer.
+
+The overall TableScan algorithm process in Normal Mode consists of the following steps:
+
+1. Read data: separate data streams are created in the Delta layer and Stable layer to read the respective data.
+2. Sort Merge: the data streams created in step 1 are sorted and merged, and the data is returned in (handle, version) order.
```
Comment (ran-huang) [inline develop/dev-guide-read-in-fast-mode.md:56]:
```suggestion
2. Sort Merge: the data streams created in Step 1 are sorted and merged, and the data is returned in (handle, version) order.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+3. Range Filter: according to the data range, filter the data generated in step 2, and then return the data.
```
Comment (ran-huang) [inline develop/dev-guide-read-in-fast-mode.md:57]:
```suggestion
3. Range Filter: according to the data range, filter the data generated in Step 2, and then return the data.
```

**Review by ran-huang — APPROVED** [review]

File: develop/dev-guide-read-in-fast-mode.md (hunk 6 above)
Comment (hfxsd) [reply develop/dev-guide-read-in-fast-mode.md]: Refined

File: develop/dev-guide-read-in-fast-mode.md [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+
+By default, all tables are in Normal Mode. You can use the following statement to view the current table mode.
+
+```sql
+SELECT table_mode FROM information_schema.tiflash_replica WHERE table_name = 'table_name' AND table_schema = 'database_name'
+```
+
+Use the following statement to switch the corresponding table to Fast Mode.
+
+{{< copyable "sql" >}}
```
Comment (joey-yez) [inline develop/dev-guide-read-in-fast-mode.md:35]: seems strange, please check if this is correct.

File: develop/dev-guide-read-in-fast-mode.md [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+> This feature is still an experimental feature. It is NOT recommended that you use it in the production environment.
+
+This document describes how to use Fast Mode to speed up queries in Online Analytical Processing (OLAP) scenarios.
+
+TiFlash supports the following modes:
+
+- `Normal Mode`. The default mode. This mode guarantees the precision of query results and data consistency.
+- `Fast Mode`. This mode does not guarantee the accuracy of query results and data consistency, but provides more efficient query performance.
+
+Some AP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can switch the corresponding table to TiFlash's Fast Mode for querying.
```
Comment (joey-yez) [inline develop/dev-guide-read-in-fast-mode.md:19]: AP is not so formal, OLAP seems good.

Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:19]:
```suggestion
Some OLAP scenarios allow for some tolerance to the accuracy of the query results. In these cases, if you need higher query performance, you can switch the corresponding table to TiFlash's Fast Mode for querying.
```

File: develop/dev-guide-read-in-fast-mode.md [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+
+By default, all tables are in Normal Mode. You can use the following statement to view the current table mode.
+
+```sql
+SELECT table_mode FROM information_schema.tiflash_replica WHERE table_name = 'table_name' AND table_schema = 'database_name'
+```
+
+Use the following statement to switch the corresponding table to Fast Mode.
+
+{{< copyable "sql" >}}
```
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:35]:
```suggestion
```

File: sql-statements/sql-statement-set-tiflash-mode.md [hunk 25]
```diff
@@ -0,0 +1,91 @@
...
+```ebnf+diagram
+AlterTableSetTiFlashModeStmt ::=
+    'ALTER' 'TABLE' TableName 'SET' 'TIFLASH' 'MODE' mode
+```
+
+## Example
+
+Assume that the `test` table has a TiFlash replica.
+
+{{< copyable "sql" >}}
```
Comment (hfxsd) [inline sql-statements/sql-statement-set-tiflash-mode.md:36]:
```suggestion
```

File: sql-statements/sql-statement-set-tiflash-mode.md [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+
+```ebnf+diagram
+AlterTableSetTiFlashModeStmt ::=
+    'ALTER' 'TABLE' TableName 'SET' 'TIFLASH' 'MODE' mode
+```
+
+## Example
+
+Assume that the `test` table has a TiFlash replica.
+
```
Comment (hfxsd) [inline sql-statements/sql-statement-set-tiflash-mode.md:35]:
```suggestion
```

File: develop/dev-guide-read-in-fast-mode.md (hunk 23 above)
Comment (hfxsd) [reply develop/dev-guide-read-in-fast-mode.md]: OK fixed.

File: develop/dev-guide-read-in-fast-mode.md (hunk 22 above)
Comment (hfxsd) [reply develop/dev-guide-read-in-fast-mode.md]: Deleted.

File: develop/dev-guide-read-in-fast-mode.md [hunk 27]
```diff
@@ -0,0 +1,59 @@
...
+## Switch to Fast Mode
+
+By default, all tables are in Normal Mode. You can use the following statement to view the current table mode.
+
+```sql
+SELECT table_mode FROM information_schema.tiflash_replica WHERE table_name = 'table_name' AND table_schema = 'database_name'
+```
+
+Use the following statement to switch the corresponding table to Fast Mode.
+
```
Comment (hfxsd) [inline develop/dev-guide-read-in-fast-mode.md:34]:
```suggestion
```

**Review by joey-yez — APPROVED** [review]

**Pull request merged**